application/core/LSYii_Validators.php

```diff
@@ -115,7 +115,9 @@ public function xssFilter($value)
         $filter = LSYii_HtmlPurifier::getXssPurifier();
 
         /** Start to get complete filtered value with  url decode {QCODE} (bug #09300). This allow only question number in url, seems OK with XSS protection **/
-        $sFiltered = preg_replace('#%7B([a-zA-Z0-9\.]*)%7D#', '{$1}', $filter->purify($value));
+        $sFiltered = $filter->purify($value);
+        $sFiltered = preg_replace('#%7B([a-zA-Z0-9\.]*)%7D#', '{$1}', $sFiltered);
+        $sFiltered = str_replace("html_entity_decode", "", $sFiltered);
```

Comment:
Need comment

Reply:
Agree. Feel free to link to the 17526 issue with a @see tag.

Reply:
Also, shouldn't this only apply to expressions? It should still be possible to include "html_entity_decode" as a normal word.

Reply:
Well, yes, but I believe we don't have EM filtering, do we?
I mean, detecting keywords inside expression is not straightforward. Possible, but not straightforward and could be very error prone.

Taking into account what you say, maybe we should just disable the internal code for html_entity_decode, and have the EM function to return as output what it receives. That is not the fancier, but could be effective.

Thoughts?

Reply:
We already work inside Expression manager :  https://github.com/LimeSurvey/LimeSurvey/blob/3d0a6091c9ec0af96a8b5cb97688bff20bb7926f/application/core/LSYii_Validators.php#L142 

Maybe here it's the function ?

Reply:
Yes, probably that would be a starting point. Will review it. 

> Taking into account what you say, maybe we should just disable the internal code for html_entity_decode, and have the EM function to return as output what it receives. That is not the fancier, but could be effective.

This wouldn't work, sorry. It wou;dn't be filtering, but disabling the whole function for everyone.